platform/ios/src/MGLMapView.mm

```diff
@@ -2909,6 +2912,23 @@ - (void)addAnnotations:(NS_ARRAY_OF(id <MGLAnnotation>) *)annotations
             _annotationContextsByAnnotationTag[annotationTag] = context;
         }
     }
+   
+   
+    MGLAnnotationContainerView *newAnnotationContainerView;
+    if (self.annotationContainerView)
+    {
+        // reload any previously added views
+        newAnnotationContainerView = [MGLAnnotationContainerView annotationContainerViewWithAnnotationContainerView:self.annotationContainerView];
+        [self.annotationContainerView removeFromSuperview];
+    }
+    else
+    {
+        newAnnotationContainerView = [[MGLAnnotationContainerView alloc] initWithFrame:self.bounds];
+    }
+    newAnnotationContainerView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
```

Comment:
Also set the `contentMode` to `UIViewContentModeCenter` like we do for the GLKView. That way the content remains centered correctly as the device rotates and probably in some other edge cases as well.